### DISCUSSION TITLE
Question about TESTHESS

### DISCUSSION THREAD

**Discussion by jacekkozuch** [TinkerTools/tinker-gpu#221] [Q&A]
Hi,

I was playing around with TESTHESS and I'm not sure if I discovered an issue or if I'm misunderstanding something.

First, when I use it in "analytical" mode I obtain the list Diagonal Hessian Elements. However, with "numerical" the program does something but there's no output at the end. However, when using both, then I get the comparison, so the numerical hessian elements are computed. Is this intended to not display the Diagonal Hessian Elements with the numerical approach?

Second (and not an issue), of course TESTHESS is slower than TESTGRAD. Is there therefore a plan transfer it also to the GPU code?

Thanks!
Jacek

**Comment by zhi-wang** [comment]
Implementing the analytical hessian for the valence terms is possible for GPU. But for multipole, polarization, and PME... 😭

**Reply by jacekkozuch** [reply]
Haha, the multipole and polarization is what I'd be interested in :P I'll just try a few more cores ;)

Thanks for the quick response!
